scripts/lib/utils.ts

```diff
@@ -0,0 +1,163 @@
+import { strict as assert } from "assert";
+import { readFileSync, existsSync } from "fs";
+import { resolve, relative } from "path";
+
+import chalk from "chalk";
+import { addMinutes, formatISO } from "date-fns";
+import faker from "faker";
+import JSON5 from "json5";
+
+import { log } from "./log";
+import { SimConfig } from "./types";
+
+const INDEX_PADDING = 4;
+const INDEX_BASE = 10 ** INDEX_PADDING + 1;
+
+// @see EmojiReactionType in types/reactions
+// noinspection SpellCheckingInspection
+const REACTIONS = [
+  "heart",
+  "clap",
+  "wolf",
+  "laugh",
+  "thatsjazz",
+  "boo",
+  "burn",
+  "sparkle",
+  "messageToTheBand",
+];
+
+const TEXTERS = [
+  faker.hacker.phrase,
+  faker.company.catchPhrase,
+  faker.company.bs,
+  faker.lorem.sentence,
+  faker.random.words,
+];
+
+export const pickFrom: <T>(array: T[]) => T | undefined = (array) =>
+  array[Math.floor(Math.random() * array.length)];
+
+export const generateRandomReaction = () => pickFrom(REACTIONS);
+export const generateRandomText = () => pickFrom(TEXTERS)?.();
+
+// export const STOP = Symbol("stop");
+// export const withLoop: (tick: number, fn: Function) => Function = (
+//   tick,
+//   fn
+// ) => {
+//   const loop = async () => {
+//     const signal = fn();
+//     if (signal !== STOP) {
+//       setTimeout(loop, tick);
+//     }
+//   };
+//   return loop;
+// };
```

Comment:
This could possibly be a utility function that takes any other function and creates a "looper" i.e. a function that uses `setTimeout()` to call itself after a delay thus doing similar thing as `setInterval()` with the added bonus of not pushing unnecessary slots for execution like `setInterval()` does and having more spaced out invocations.